File: packages/ts-sdk/test/adapters/knowledge/graph.test.ts
import { describe, expect, it } from 'vitest';
import { GraphAdapter, TREEDX_OPENAPI_OPERATIONS } from '../../../src/treedx/index.js';
import { MockTransport } from '../mock.js';

describe('GraphAdapter', () => {
  it('declares every capability required to build the graph from repository content', () => {
    expect(TREEDX_OPENAPI_OPERATIONS.find(operation => operation.operationId === 'refreshRepositoryGraph')?.requiredCapabilities)
      .toEqual(['files:read', 'git:read', 'graph:refresh']);
  });
  it('constructs graph endpoints', async () => {
    const transport = new MockTransport();
    const adapter = new GraphAdapter({ transport });
    await adapter.refresh('repo');
    await adapter.query('repo', {});
    expect(transport.requests.map((r) => `${r.method} ${r.path}`)).toEqual([
      'POST /api/v1/repos/repo/graph/refresh',
      'POST /api/v1/repos/repo/graph/query'
    ]);
  });
});

File: packages/elixir-sdk/test/adapters/knowledge/graph_test.exs
defmodule TreeDxSdk.GraphAdapterTest do
  use ExUnit.Case, async: true

  test "all constructs expected request" do
    {:ok, pid} = TreeDxSdk.Test.MockTransport.start_link()
    client = TreeDxSdk.Test.MockTransport.client(pid)
    TreeDxSdk.Graph.refresh(client, "repo/a")
    TreeDxSdk.Graph.query(client, "repo/a", %{})

    assert Enum.any?(
             TreeDxSdk.Test.MockTransport.requests(pid),
             &(&1.method == :post and &1.path == "/api/v1/repos/repo%2Fa/graph/query")
           )
  end
end
